Sources/SpeziOnboarding/OnboardingIdentifiableViewModifier.swift

```diff
@@ -0,0 +1,60 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+/// Wrap `Content` `View` in an `Identifiable` `View`.
+private struct OnboardingIdentifiableView<Content, ID>: View, Identifiable where Content: View, ID: Hashable {
+    /// Unique identifier of the wrapped `View`.
+    let id: ID
+    /// Wrapped `View`.
+    let body: Content
+}
+
+private struct OnboardingIdentifiableViewModifier<ID>: ViewModifier, Identifiable where ID: Hashable {
+    let id: ID
+
+    func body(content: Content) -> some View {
+        OnboardingIdentifiableView(
+            id: self.id,
+            body: content
+        )
```

Comment:
Something I noticed just now, I think it isn't necessary to wrap this into a `OnboardingIdentifiableView` anymore and we can remove the view above, as the ViewModifier now conforms to `Identifiable`.